model/src/main/kotlin/Project.kt

```diff
@@ -155,7 +155,25 @@ data class Project(
     /**
      * A comparison function to sort projects by their identifier.
      */
-    override fun compareTo(other: Project) = id.compareTo(other.id)
+    override fun compareTo(other: Project): Int {
+        // First sort by package type.
+        var sign = id.type.compareTo(other.id.type)
+
+        if (sign == 0) {
```

Comment:
Consider returning early here to eliminate some nesting.

Reply:
Done.